### ISSUE TITLE
AH01786: uri mismatch - </repo.git/info/refs> does not match request-uri </repo.git/info/refs?service=git-upload-pack>

### ISSUE THREAD

**Issue by sgrossberndt** [git-for-windows/git#1979]
### Setup

Git 2.19.2 and 2.20.0 64bit

```
$ git --version --build-options
git version 2.19.2.windows.1
cpu: x86_64
built from commit: 26dcaa1b6b5fd862db3ec40983e33ff3432f1166
sizeof-long: 4
sizeof-size_t: 8
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.17134.407]
```

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

CMD

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git.exe clone -v "https://host/repo.git" "C:\Users\MyUser\Desktop\repo"
```
 - What did you expect to occur after running these commands?

Cloned repo.

 - What actually happened instead?

```
fatal: unable to access 'https://host/repo.git/': The requested URL returned error: 400
```

In the server log: 
```
[Wed Dec 12 13:00:29.685479 2018] [auth_digest:error] [pid 1350:tid 139845021583104] [client x.x.x.x:11502] AH01786: uri mismatch - </repo.git/info/refs> does not match request-uri </repo.git/info/refs?service=git-upload-pack>
```

I think I already found the underlying issue - the upgrade of cURL v7.62.0. There is an according issue for that behaviour in https://github.com/curl/curl/issues/3353 which has been fixed there and is included in v.7.63.0 which was released 5 hours ago.

Downgrading to 2.19.1 solves the issue for me.

**Comment by dscho** [comment]
I just upgraded to cURL v7.63.0 which purportedly has a fix for this type of issue. Could you try with `libcurl-4.dll` from https://github.com/git-for-windows/git-sdk-64/blob/master/mingw64/bin/libcurl-4.dll?

**Comment by sgrossberndt** [comment]
Thanks for the fast response! I had to replace the file `libcurl-4.dll` in `C:\Program Files\Git\mingw64\libexec\git-core` and `C:\Program Files\Git\mingw64\bin` - now it works

**Comment by dscho** [comment]
This will be fixed in v2.20.1, which I get the impression is imminent (Git v2.20.1, that is, Git for Windows is always released as soon afterward as possible).
